imap_processing/cdf/utils.py

```diff
@@ -14,7 +15,9 @@
 logger = logging.getLogger(__name__)
 
 
-def calc_start_time(shcoarse_time: float) -> np.datetime64:
+def calc_start_time(
+    shcoarse_time: float, launch_time: Optional[np.datetime64] = launch_time
```

Comment:
No need to change this here, but I find it odd we have `launch_time` in the root level of `imap_processing`. There are conflicting uses of it as well with IDEX also using a different value, and `hi` the only instrument using this value outside of this helper function.
https://github.com/search?q=repo%3AIMAP-Science-Operations-Center%2Fimap_processing%20launch_time&type=code

Reply:
I thought I updated all places it uses it. Looks like I missed IDEX. I started seeing that we were storing launch_time in different places and thought we should keep it in one place and update it when we get real one.

Is there reason for making this argument optional or adding it? Just curious

Reply:
My motivation for adding the parameter is because CoDICE prefers to use a `launch_time` of `2010-01-01` instead of `2010-01-01T00:01:06.184` 🤷🏻 

Alternatively, I could just take care of this within the CoDICE code by essentially making a CoDICE-specific `calc_start_time` function.

Reply:
Oh I think I misunderstood you question. Are you asking whether it is best to make the parameter optional or required?

Reply:
Your first response was what I was wondering. Cool

Reply:
Maybe this shouldn't be a cdf_utils helper, but a more general helper for any time offset from launch epoch?

Right now hi is creating a `get_direct_event_time()` using the global launch time, but that is really just the same function as here. `calc_time_from_launch(offset, units="ns", launch_time=some_default)` could then be used by hi with their current default, idex with theirs, and codice with theirs.

I still think this is good to go and can be cleaned up later.

Reply:
Ah ok. That makes sense!